modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/di/ServersModule.kt

```diff
@@ -154,10 +154,15 @@ class ServersModule {
     }
 
     @Provides
-    internal fun provideOkHttpClientBuilder(@SyncServerCache cache: Cache): OkHttpClient.Builder {
+    internal fun provideOkHttpClientBuilder(
+        @SyncServerCache cache: Cache,
+        networkInterceptors: Set<@JvmSuppressWildcards Interceptor>,
+    ): OkHttpClient.Builder {
         var builder = OkHttpClient.Builder()
-            .addNetworkInterceptor(INTERCEPTOR_CACHE_MODIFIER)
-            .addNetworkInterceptor(INTERCEPTOR_USER_AGENT)
+            .apply {
+                (networkInterceptors + INTERCEPTOR_CACHE_MODIFIER + INTERCEPTOR_USER_AGENT)
+                    .forEach(::addNetworkInterceptor)
+            }
```

Comment:
This seems a bit cleaner in terms of code formatting and it doesn't need to append lists.

```suggestion
            .apply { networkInterceptors.forEach(::addNetworkInterceptor) }
            .addNetworkInterceptor(INTERCEPTOR_CACHE_MODIFIER)
            .addNetworkInterceptor(INTERCEPTOR_USER_AGENT)
```

Reply:
Thanks! Order is something I didn't consider - I don't know if it matters (interceptors are already applied) but it's maybe better to add Sentry's one as the last. Addressed in 29f12cc857a9090f93fbaa96a85f0e8812579848

Reply:
Or it _does_ matter, as interceptors are executed in order of applying?

Reply:
Ok I answered myself, it makes sense 😅 

> OkHttp uses lists to track interceptors, and interceptors are called in order. (https://square.github.io/okhttp/features/interceptors/)

So yeah, definitely it's better to add Sentry as the last one 👍

Reply:
In that case wouldn't it be better to not include Sentry interceptor in set and inject it explicitly with some named annotation? Otherwise if we include other interceptors in the set in the future Sentry one might not be added as the last one.

Reply:
Sure 👍 the injection via set wasn't the best idea here, especially as it's the only interceptor. Addressed in 6e4023aa53198c8797bb37034c20df0b14900f1c